pandas/core/arrays/datetimelike.py

```diff
@@ -2105,3 +2110,21 @@ def maybe_infer_freq(freq):
             freq_infer = True
             freq = None
     return freq, freq_infer
+
+
+def dtype_to_unit(dtype: DatetimeTZDtype | np.dtype) -> str:
+    """
+    Return the unit str corresponding to the dtype's resolution.
+
+    Parameters
+    ----------
+    dtype : DatetimeTZDtype or np.dtype
+        If np.dtype, we assume it is a datetime64 dtype.
+
+    Returns
+    -------
+    str
+    """
+    if isinstance(dtype, DatetimeTZDtype):
+        return dtype.unit
+    return str(dtype).split("[")[-1][:-1]
```

Comment:
`datetime_data` is probably the more "official" way to get the unit from an np.dtype: https://numpy.org/devdocs/reference/generated/numpy.datetime_data.html